frontend/src/pages/ArtifactDetails.tsx

```diff
@@ -153,4 +167,15 @@ export default class ArtifactDetails extends Page<{}, ArtifactDetailsState> {
   private switchTab(selectedTab: number) {
     this.setState({selectedTab});
   }
+
+  private static buildResourceDetailsPageRoute(
+    resource: LineageResource, typeName: string): string {
+    let route;
+    if (resource instanceof Artifact) {
+      route = RoutePage.ARTIFACT_DETAILS.replace(`:${RouteParams.ARTIFACT_TYPE}+`, typeName);
+    } else {
+      route = RoutePage.EXECUTION_DETAILS.replace(`:${RouteParams.EXECUTION_TYPE}+`, typeName);
+    }
+    return route.replace(`:${RouteParams.ID}`, String(resource.getId()));
+  }
```

Comment:
Just a suggestion for brevity, entire function can be:
```js
private static buildResourceDetailsPageRoute(resource: LineageResource, typeName: string): string {
    const tp = resource instanceof Artifact ? 'ARTIFACT' : 'EXECUTION'
    const route = RoutePage[`${tp}_DETAILS`]
        .replace(`:${RouteParams[`${tp}_TYPE`]}+`, typeName)
        .replace(`:${RouteParams.ID}`, String(resource.getId()));
    return route;
}
```

Reply:
Simplified a bit, but I prefer readability over brevity and the ability for tooling to understand what ROUTE_PARAMS.KEY means.